cmd/ndm_daemonset/probe/eventhandler.go

```diff
@@ -141,19 +142,20 @@ func (pe *ProbeEvent) deleteBlockDeviceEvent(msg controller.EventMessage) {
 
 	// rescan only if GPT based UUID is disabled.
 	if !isDeactivated && !isGPTBasedUUIDEnabled {
-		go pe.initOrErrorEvent()
+		go pe.Rescan()
 	}
 }
 
-// initOrErrorEvent rescan system and update disk resource this is
-// used for initial setup and when any uid mismatch or error occurred.
-func (pe *ProbeEvent) initOrErrorEvent() {
+// Rescan syncs etcd and NDM
+func (pe *ProbeEvent) Rescan() error {
```

Comment:
@harshthakur9030  Does it make sense to move Rescan out of this and keep it completely in the udevprobe. That way you tie the rescan with udev,

Reply:
I think it's better if we keep it in event handler itself as a probe event method as probe event struct is simpler to initialize than a udev probe struct(constructor for which isn't exported) so considering Rescan is actually just a wrapper around the udev scan, it's fine I suppose.

Reply:
`udev.Rescan(c Controller)`, something like this, because we have the controller. Initialising both Probevent and rescan only requires the pointer to controller.

Reply:
Makes sense. Will refactor this then